app/views/users/access_list.html.haml

```diff
@@ -24,21 +24,26 @@
           %tr
             %th.approval-column= Product.human_attribute_name(:can_be_used_by?)
             %th.product-column= Product.model_name.human
+            %th.product-column= ProductUser.human_attribute_name(:date_added)
             %th.scheduling-group-column= ProductAccessGroup.model_name.human
 
         %tbody
           - products.each do |product|
+            - access_granted_at = @access_granted_date_by_product_id[product.id]
```

Comment:
I'd vote for using the `find_product_user` method instead:
```suggestion
            - access_granted_at = product.find_product_user(@user)
```
I think this is a little bit simpler, and wouldn't require any changes to the controller logic.  What do you think?